lib/internal/main/test_runner.js

```diff
@@ -22,7 +28,43 @@ if (isUsingInspector()) {
   inspectPort = process.debugPort;
 }
 
-run({ concurrency, inspectPort, watch: getOptionValue('--watch'), setup: setupTestReporters })
+let shard;
+const shardOption = getOptionValue('--test-shard');
+if (shardOption) {
+  const shardParts = shardOption.split('/');
+
+  if (shardParts.length !== 2) {
+    process.exitCode = kGenericUserError;
+
+    throw new ERR_INVALID_ARG_VALUE(
+      '--test-shard',
+      shardOption,
+      'must be in the form of <index>/<total>',
+    );
+  }
+
+  const { 0: indexStr, 1: totalStr } = shardParts;
+
+  const index = NumberParseInt(indexStr);
+  const total = NumberParseInt(totalStr);
+
+  if (NumberIsNaN(index) || NumberIsNaN(total)) {
```

Comment:
I think we should be stricter than that. Currently, because of how `Number.parseInt` works this would be accepted: `0xf/20abcd`. Maybe use a regex to validate the format?

Reply:
Yeah, I thought so too but regex can be more problematic...

will update

Reply:
done